src/main/java/org/jenkinsci/plugins/stashNotifier/StashNotifier.java

```diff
@@ -636,21 +672,50 @@ private HttpEntity newStashBuildNotificationEntity(
         json.put("description", getBuildDescription(build, state));
         json.put("url", Jenkins.getInstance()
         		.getRootUrl().concat(build.getUrl()));
-        
+
         return new StringEntity(json.toString(), "UTF-8");
 	}
 
+
+
 	/**
-	 * Returns the build key used in the Stash notification. Includes the 
+	 * Returns the build key used in the Stash notification. Includes the
 	 * build number depending on the user setting.
-	 * 
+	 *
 	 * @param 	build	the build to notify Stash of
 	 * @return	the build key for the Stash notification
 	 */
-	private String getBuildKey(final AbstractBuild<?, ?> build) {
-		StringBuilder key = new StringBuilder();
+	private String getBuildKey(final AbstractBuild<?, ?> build,
+							   BuildListener listener) {
+
+        String overriddenKey = (projectKey != null && projectKey.trim().length() > 0) ? projectKey : getDescriptor().getProjectKey();
```

Comment:
I'm not sure it is the best way to check that parameter is not empty. Is this correct? `projectKey` is never set up at global configuration, so I can miss something due to I'm new to Jenkins API